chains/solana/contracts/programs/ccip-router/src/state.rs

```diff
@@ -20,9 +22,33 @@ pub struct Config {
     pub fee_aggregator: Pubkey, // Allowed address to withdraw billed fees to (will use ATAs derived from it)
 }
 
-#[derive(Clone, AnchorSerialize, AnchorDeserialize, InitSpace, Debug)]
+#[derive(Clone, AnchorSerialize, AnchorDeserialize, InitSpace, Debug, PartialEq, Eq, Copy)]
+pub enum RestoreOnAction {
+    None,
+    Upgrade,
+    Rollback,
+}
```

Comment:
I don't think there's necessarily anything wrong with this naming, but the logic here takes a moment to understand, so I think comments would be good.